gobblin-utility/src/main/java/gobblin/util/SchedulerUtils.java

```diff
@@ -178,6 +215,96 @@ public boolean accept(File file) {
   }
 
   /**
+   * Recursively load job configuration files under given URI of **directory of config files folder**
+   * todo
+   */
+  private static void loadGenericJobConfigsRecursive(List<Properties> jobConfigs, Properties rootProps,
+      Set<String> jobConfigFileExtensions, URI uri)
+      throws ConfigurationException, IOException {
+
+    /* create the generic file system */
+    org.apache.hadoop.conf.Configuration conf = new org.apache.hadoop.conf.Configuration();
+    FileSystem filesystem = FileSystem.get(uri, conf);
+
+    Path configDirPath = new Path(uri);
+    if (!filesystem.exists(configDirPath)) {
+      throw new RuntimeException("No specified .properties directory found :" + configDirPath.toString());
+    }
+
+    FileStatus[] propertiesFilesStatus = filesystem.listStatus(configDirPath, PROPERTIES_PATH_FILTER);
+    ArrayList<String> propertiesFiles = new ArrayList<>();
```

Comment:
There is no reason to use a `List` here. You could just do `filesystem.open(propertiesFilesStatus[0].getPath())` (after checking for uniqueness).